metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/store/repos/PolarisTableCustomRepository.java

```diff
@@ -12,8 +11,9 @@ public interface PolarisTableCustomRepository {
      * @param dbName database name
      * @param tableNamePrefix table name prefix. can be empty.
      * @param pageSize target size for each page
+     * @param selectAll if true return the PolarisEntity else return name of the entity
      * @return table entities in the database.
      */
-    List<PolarisTableEntity> findAllTablesByDbNameAndTablePrefix(
-        String dbName, String tableNamePrefix, int pageSize);
+    List<?> findAllTablesByDbNameAndTablePrefix(
+        String dbName, String tableNamePrefix, int pageSize, boolean selectAll);
```

Comment:
The `selectAll` name could be improved since the method name already has `findAll` so it could seem like it might not return the whole list, this could be renamed instead to `selectNames` or something similar